contracts/src/token/erc1155/extensions/metadata_uri.rs

```diff
@@ -0,0 +1,31 @@
+//! Optional MetadataURI of the ERC-1155 standard.
+
+use alloc::string::String;
+
+use stylus_proc::{public, sol_storage};
+
+sol_storage! {
+    /// MetadataURI of an [`crate::token::erc1155::Erc1155`] token.
+    pub struct Erc1155MetadataURI {
+        /// Optional base URI for tokens.
+        string _base_uri;
+    }
+}
+
+/// Interface for the optional metadata functions from the ERC-1155 standard.
+pub trait IErc1155MetadataURI {
+    /// Returns the base of Uniform Resource Identifier (URI) for tokens'
+    /// collection.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Read access to the contract's state.
+    fn base_uri(&self) -> String;
+}
+
+#[public]
+impl IErc1155MetadataURI for Erc1155MetadataURI {
```

Comment:
You're going to expose selector `baseUri()` to public that doesn't reflect solidity currently. We're having the same thing for erc721 currently and gonna remove it in nearest future.